### PR TITLE
lot of change in Chat 

### PR DESCRIPTION
**Pull request by emmanuel-lbs**
- on chanels members list we can click on user for open menu: Profile - Invite Game - DM - Ban - Mute (only Profile work for now)
Screenshot from 2023-02-18 14-19-20

- i change the Chat Onglets Tabs/Tablist/Tabpanel react by my homemade onglets (il faut du css c'est pas beau la ..)
Screenshot from 2023-02-18 14-23-38

- i withdrew the text-input in Chanels and Dm if where are not in a Chanel or a Dm  
- i upgrade Chat/Chanels  routes  and  create route for Dm pages ( so we can go on Dm now) 

on the back : 
- changes on DTO
- We can unMute and addAdmin

-----------------
Front:
- tried changing colors to match the rest of the project (to avoid being Discord copy)
  but i proved myself of having a bad taste and still don't like the colors i changed :rofl: 
- applied private channel mode when creating a channel + a little lock icon appears beside the private channel name
- basic conversation on channel finally works with correct sender info (avatar, name). 

[future todos]
- do more tests with channels, handle protected channel, do (fix) front parts that need to be done

### CODE REVIEW

**Review by KobeLaC — APPROVED** [review]
On my side (chat - back ) :
- changes on DTO
- unMute
- addAdmin 
- test timer on mute/ban

**Review by Acusanno — APPROVED** [review]
Everything looks alright.

File: back/src/chat/channel/entities/channel.entity.ts [hunk 1]
```diff
@@ -5,7 +5,6 @@ import { Column, Entity, JoinTable, ManyToMany, ManyToOne, OneToMany, OneToOne,
 enum ChanType {
 	Public,
 	Private,
```
Comment (se-lee) [inline back/src/chat/channel/entities/channel.entity.ts:7]: I think we should put back protected --- I thought Protected was optional and we could do Public and Private OR Protected, but according to Simon, Private and Protected channels are two different things , Private can only be joined with an invite (password is not mandatory) and Protected can be joined without an invite but with a password (so anyone who knows the password can join without an invite).

**Review by se-lee — APPROVED** [review]
Thanks manu for taking care of chat front :100:  you are too great manu 
i didnt know what to do with router but you saved my life :1st_place_medal:

**Pull request merged**